src/main/java/com/merge/api/resources/accounting/companyinfo/CompanyInfoClient.java

```diff
@@ -89,8 +90,13 @@ public PaginatedCompanyInfoList list(CompanyInfoListRequest request, RequestOpti
                 .addHeader("Content-Type", "application/json");
         Request okhttpRequest = _requestBuilder.build();
         try {
-            Response response =
-                    clientOptions.httpClient().newCall(okhttpRequest).execute();
+            OkHttpClient client = clientOptions.httpClient();
+            if (requestOptions.getTimeout().isPresent()) {
```

Comment:
Hey @rmkonnur 👋 

Glad to see we can override the okHttpClient timeout with the `requestOptions` 💪 
Yet would it be possible to be retro-compatible..?

Before I was doing this call ⬇️ 

```java
    List<CompanyInfo> companies =
        accountingClient
            .companyInfo()
            .list(CompanyInfoListRequest.builder().pageSize(1).remoteId(externalId).build())
            .getResults()
            .orElse(List.of());
```

Which was under the hood calling `list(request, null)`

---

Now it crashes because the case where requestOptions is `null` is not properly handled.

```bash
Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: java.lang.NullPointerException: Cannot invoke "com.merge.api.core.RequestOptions.getTimeout()" because "requestOptions" is null] with root cause 
java.lang.NullPointerException: Cannot invoke "com.merge.api.core.RequestOptions.getTimeout()" because "requestOptions" is null
        at com.merge.api.resources.accounting.companyinfo.CompanyInfoClient.list(CompanyInfoClient.java:94)
```

PS: this applies to a lot of other clients as well as `retrieve` / `list`  methods

---

Any chance you patch this and let your sdk handle this smoothly?

Reply:
Hi @pastouret, Thank you for raising this issue. I have just released `v1.0.7` of the merge-java-client which should include null handling for the `requestOptions` field. Feel free to follow up in a Github Issue if there continue to be issues with this!

Reply:
@rmkonnur many thanks for the lightening fast support, it works like a charm 🙏